src/components/views/rooms/RoomList.js

```diff
@@ -236,13 +234,6 @@ module.exports = React.createClass({
         this._updateStickyHeaders(true, scrollToPosition);
     },
 
-    onRoomTimeline: function(ev, room, toStartOfTimeline, removed, data) {
-        if (toStartOfTimeline) return;
-        if (!room) return;
-        if (data.timeline.getTimelineSet() !== room.getUnfilteredTimelineSet()) return;
-        this._delayedRefreshRoomList();
```

Comment:
These checks seem to have disappeared, as well as the _delayedRefreshRoomList?

Reply:
So:
 - the `_delayedRefreshRoomList` is done when we get an update from RoomListStore;
 - [ ] `toStartOfTimeline` ... I forgot about, and we do get it in the event so should be an easy fix;
 - [ ] `!room` ... I also forgot about but find it hard to imagine an event that doesn't have a room. I suppose it's strictly possible.
 - the timeline set check I thought would be handle by `isLiveEvent`? Unless I've got that confused?

Reply:
Ah right, OK. I think this almost all true apart from the timeline set check, where live timeline !== unfiltered timeline set. There are multiple timeline sets, the first is the unfiltered one (the normal one that isn't a search result). Each timeline set has many timelines, one of which is live. `liveEvent` is telling you if it's to the live timeline in a timeline set, not if it's to the timeline set you're interested in.

Reply:
So in theory we shouldn't get live events for search result or any other timeline set apart from the unfiltered one. We're not sure how true this is though. Investigation of exactly when Room.timeline is emitted will have to be done.

Reply:
Seemingly this isn't true, as @dbkr has said, each set (be it a room timeline, notification timeline, rooms timeline) has a live timeline, and the only way of us knowing which is the *room's* main timeline if by comparing references.